src/main/java/io/jenkins/plugins/sample/OnboardingConfiguration.java

```diff
@@ -224,4 +280,177 @@ public FormValidation doTestConnection(
+        return CredentialsMatchers.firstOrNull(
+                CredentialsProvider.lookupCredentialsInItem(StandardCredentials.class, null, ACL.SYSTEM2, null),
+                CredentialsMatchers.withId(credentialId));
+    }
+
+    public String getLatestJobPerCategory() {
+
+        String perCategoryJobNames = Jenkins.get().getRootDir().getAbsolutePath() + "/latestCategoryJobs.txt";
+        Path perCategoryJobNamesFilepath = Paths.get(perCategoryJobNames);
+        Map<String, String> categoryMap = new HashMap<>();
+        StringBuilder jobsPerCategory = new StringBuilder();
+
+        if (Files.exists(perCategoryJobNamesFilepath)) {
+            try (Stream<String> lines = Files.lines(perCategoryJobNamesFilepath)) {
+                lines.forEach(line -> {
+                    String[] parts = line.split(":", 2);
+                    if (parts.length == 2) {
+                        categoryMap.put(parts[0].trim(), parts[1].trim());
+                    }
+                });
+            } catch (IOException e) {
+                return "Error reading category job records.";
+            }
+        } else {
+            return "The file does not exist or cannot be accessed.";
+        }
+
+        if (categoryMap.isEmpty()) {
+            return "No job names found for any category.";
+        } else {
+            for (Map.Entry<String, String> entry : categoryMap.entrySet()) {
+                jobsPerCategory
+                        .append("Category: ")
+                        .append(entry.getKey())
+                        .append(", Latest Job: ")
+                        .append(entry.getValue())
+                        .append("<br/>");
+            }
+        }
+        return jobsPerCategory.toString();
+    }
+
+    @POST
+    public FormValidation doRenameJob(
+            @QueryParameter("oldJobName") final String oldJobName,
+            @QueryParameter("newJobName") final String newJobName) {
+        try {
+            Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+            String jenkinsBaseUrl = System.getProperty("JENKINS_BASE_URL");
+            String username = System.getProperty("JENKINS_USERNAME");
+            String apiToken = System.getProperty("JENKINS_API_TOKEN");
+            if (Util.fixEmptyAndTrim(jenkinsBaseUrl) == null
+                    || Util.fixEmptyAndTrim(apiToken) == null
+                    || Util.fixEmptyAndTrim(username) == null
+                    || Util.fixEmptyAndTrim(oldJobName) == null
+                    || Util.fixEmptyAndTrim(newJobName) == null) {
+                return FormValidation.error("oldJobName or newJobName must not be null or empty ");
+            }
+            String queryString = URLEncoder.encode(newJobName.trim(), StandardCharsets.UTF_8);
+            String renameUrl =
+                    String.format("%s/job/%s/doRename?newName=%s", jenkinsBaseUrl, oldJobName.trim(), queryString);
```

Comment:
This is indeed a way to rename a job using the REST API. However are you forced to use the REST API? 
You're already inside Jenkins, you could try to call this `doRename` method directly.